Blazor project template fails to build: Unable to find package Microsoft.AspNetCore.SpaServices.Extensions

### Repro steps:
Install Visual Studio 2017 15.6.2
Install the latest .NET Core SDK nightly from https://github.com/dotnet/cli
Install Blazor extension from MyGet
Create Blazor app

### Result:
Restore failed, but oddly enough the build process succeeded.

### Here is the Output list:
NU1101: Unable to find package Microsoft.AspNetCore.SpaServices.Extensions. No packages exist with this id in source(s): https://dotnet.myget.org/f/blazor-dev/api/v3/index.json
NuGet package restore failed. Please see Error List window for detailed warnings and errors.

#### After rebuilding:
1>------ Rebuild All started: Project: WebApplication1, Configuration: Debug Any CPU ------
1>C:\git\BlazerLab\WebApplication1\WebApplication1\WebApplication1.csproj : warning MSB4241: The SDK reference "Microsoft.NET.Sdk.Razor" version "2.1.0-preview2-30230" was resolved to version "2.1.300-preview2-008354" instead.  You could be using a different version than expected if you do not update the referenced version to match.
1>C:\git\BlazerLab\WebApplication1\WebApplication1\WebApplication1.csproj : warning MSB4240: Multiple versions of the same SDK "Microsoft.NET.Sdk.Razor" cannot be specified. The SDK version "2.1.300-preview2-008354" already specified by "C:\git\BlazerLab\WebApplication1\WebApplication1\WebApplication1.csproj" will be used and the version "2.1.0-preview2-30230" will be ignored.
1>C:\git\BlazerLab\WebApplication1\WebApplication1\WebApplication1.csproj : warning MSB4240: Multiple versions of the same SDK "Microsoft.NET.Sdk.Razor" cannot be specified. The SDK version "2.1.300-preview2-008354" already specified by "C:\git\BlazerLab\WebApplication1\WebApplication1\WebApplication1.csproj" will be used and the version "2.1.0-preview2-30230" will be ignored.
1>C:\Program Files\dotnet\sdk\2.1.300-preview2-008354\Sdks\Microsoft.NET.Sdk.Web\Sdk\Sdk.props(25,3): warning MSB4011: "C:\Program Files\dotnet\sdk\2.1.300-preview2-008354\Sdks\Microsoft.NET.Sdk\Sdk\Sdk.props" cannot be imported again. It was already imported at "C:\Program Files\dotnet\sdk\2.1.300-preview2-008354\Sdks\Microsoft.NET.Sdk.Razor\Sdk\Sdk.props (18,3)". This is most likely a build authoring error. This subsequent import will be ignored. [C:\git\BlazerLab\WebApplication1\WebApplication1\WebApplication1.csproj]
1>C:\Program Files\dotnet\sdk\2.1.300-preview2-008354\Sdks\Microsoft.NET.Sdk.Web\Sdk\Sdk.props(27,3): warning MSB4011: "C:\Program Files\dotnet\sdk\2.1.300-preview2-008354\Sdks\Microsoft.NET.Sdk.Razor\Sdk\Sdk.props" cannot be imported again. It was already imported at "C:\git\BlazerLab\WebApplication1\WebApplication1\WebApplication1.csproj". This is most likely a build authoring error. This subsequent import will be ignored.
1>C:\git\BlazerLab\WebApplication1\WebApplication1\WebApplication1.csproj : warning MSB4240: Multiple versions of the same SDK "Microsoft.NET.Sdk.Razor" cannot be specified. The SDK version "2.1.300-preview2-008354" already specified by "C:\git\BlazerLab\WebApplication1\WebApplication1\WebApplication1.csproj" will be used and the version "2.1.0-preview2-30230" will be ignored.
1>C:\Program Files\dotnet\sdk\2.1.300-preview2-008354\Sdks\Microsoft.NET.Sdk.Web\Sdk\Sdk.targets(14,3): warning MSB4011: "C:\Program Files\dotnet\sdk\2.1.300-preview2-008354\Sdks\Microsoft.NET.Sdk\Sdk\Sdk.targets" cannot be imported again. It was already imported at "C:\Program Files\dotnet\sdk\2.1.300-preview2-008354\Sdks\Microsoft.NET.Sdk.Razor\Sdk\Sdk.targets (14,3)". This is most likely a build authoring error. This subsequent import will be ignored. [C:\git\BlazerLab\WebApplication1\WebApplication1\WebApplication1.csproj]
1>C:\Program Files\dotnet\sdk\2.1.300-preview2-008354\Sdks\Microsoft.NET.Sdk.Web\Sdk\Sdk.targets(16,3): warning MSB4011: "C:\Program Files\dotnet\sdk\2.1.300-preview2-008354\Sdks\Microsoft.NET.Sdk.Razor\Sdk\Sdk.targets" cannot be imported again. It was already imported at "C:\git\BlazerLab\WebApplication1\WebApplication1\WebApplication1.csproj". This is most likely a build authoring error. This subsequent import will be ignored.
1>You are working with a preview version of the .NET Core SDK. You can define the SDK version via a global.json file in the current project. More at https://docs.microsoft.com/en-us/dotnet/core/tools/global-json
1>WebApplication1 -> C:\git\BlazerLab\WebApplication1\WebApplication1\bin\Debug\netstandard2.0\WebApplication1.dll
1>Writing index to: C:\git\BlazerLab\WebApplication1\WebApplication1\obj\Debug\netstandard2.0\blazor\index.html
1>Blazor Build result -> 37 files in C:\git\BlazerLab\WebApplication1\WebApplication1\bin\Debug\netstandard2.0\dist
========== Rebuild All: 1 succeeded, 0 failed, 0 skipped ==========

#### Please note that the rebuild process succeeded!
This is really odd, why the build succeeded if the NuGet restore failed?

#### In fact, starting with F5, the browser is launched but after a small wait we receive an error:
HTTP Error 502.5 - Process Failure

--- a/Blazor.sln
+++ b/Blazor.sln
@@ -69,6 +69,9 @@ EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "BlazorStandalone.CSharp", "src\Microsoft.AspNetCore.Blazor.Templates\content\BlazorStandalone.CSharp\BlazorStandalone.CSharp.csproj", "{A092FA91-856B-4ACA-B1C2-10BDBA366185}"
 EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "BlazorHosted.CSharp", "BlazorHosted.CSharp", "{73DA1DFD-79F0-4BA2-B0B6-4F3A21D2C3F8}"
+	ProjectSection(SolutionItems) = preProject
+		src\Microsoft.AspNetCore.Blazor.Templates\content\BlazorHosted.CSharp\NuGet.config = src\Microsoft.AspNetCore.Blazor.Templates\content\BlazorHosted.CSharp\NuGet.config
+	EndProjectSection
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "BlazorHosted.CSharp.Client", "src\Microsoft.AspNetCore.Blazor.Templates\content\BlazorHosted.CSharp\BlazorHosted.CSharp.Client\BlazorHosted.CSharp.Client.csproj", "{7549444A-9C81-44DE-AD0D-2C55501EAAC7}"
 EndProject

--- a/src/Microsoft.AspNetCore.Blazor.Templates/content/BlazorHosted.CSharp/BlazorHosted.CSharp.Client/BlazorHosted.CSharp.Client.csproj
+++ b/src/Microsoft.AspNetCore.Blazor.Templates/content/BlazorHosted.CSharp/BlazorHosted.CSharp.Client/BlazorHosted.CSharp.Client.csproj
@@ -3,9 +3,6 @@
   <PropertyGroup>
     <TargetFramework>netstandard2.0</TargetFramework>
     <OutputType>Exe</OutputType>
-
-    <!-- Temporarily, fetch Blazor packages from the blazor-dev feed -->
-    <RestoreSources>$(RestoreSources);https://dotnet.myget.org/f/blazor-dev/api/v3/index.json</RestoreSources>
   </PropertyGroup>
 
   <ItemGroup>

--- a/src/Microsoft.AspNetCore.Blazor.Templates/content/BlazorHosted.CSharp/BlazorHosted.CSharp.Server/BlazorHosted.CSharp.Server.csproj
+++ b/src/Microsoft.AspNetCore.Blazor.Templates/content/BlazorHosted.CSharp/BlazorHosted.CSharp.Server/BlazorHosted.CSharp.Server.csproj
@@ -2,9 +2,6 @@
 
   <PropertyGroup>
     <TargetFramework>netcoreapp2.0</TargetFramework>
-
-    <!-- Temporarily, fetch Blazor packages from the blazor-dev feed -->
-    <RestoreSources>$(RestoreSources);https://dotnet.myget.org/f/blazor-dev/api/v3/index.json</RestoreSources>
   </PropertyGroup>
 
   <ItemGroup>

--- a/src/Microsoft.AspNetCore.Blazor.Templates/content/BlazorHosted.CSharp/NuGet.config
+++ b/src/Microsoft.AspNetCore.Blazor.Templates/content/BlazorHosted.CSharp/NuGet.config
@@ -1,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+  <packageSources>
+    <add key="blazor-dev" value="https://dotnet.myget.org/F/blazor-dev/api/v3/index.json" />
+  </packageSources>
+</configuration>

--- a/src/Microsoft.AspNetCore.Blazor.Templates/content/BlazorStandalone.CSharp/BlazorStandalone.CSharp.csproj
+++ b/src/Microsoft.AspNetCore.Blazor.Templates/content/BlazorStandalone.CSharp/BlazorStandalone.CSharp.csproj
@@ -4,9 +4,6 @@
     <TargetFramework>netstandard2.0</TargetFramework>
     <RunCommand>dotnet</RunCommand>
     <RunArguments>blazor serve</RunArguments>
-
-    <!-- Temporarily, fetch Blazor packages from the blazor-dev feed -->
-    <RestoreSources>$(RestoreSources);https://dotnet.myget.org/f/blazor-dev/api/v3/index.json</RestoreSources>
   </PropertyGroup>
 
   <ItemGroup>

--- a/src/Microsoft.AspNetCore.Blazor.Templates/content/BlazorStandalone.CSharp/NuGet.config
+++ b/src/Microsoft.AspNetCore.Blazor.Templates/content/BlazorStandalone.CSharp/NuGet.config
@@ -1,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<configuration>
+  <packageSources>
+    <add key="blazor-dev" value="https://dotnet.myget.org/F/blazor-dev/api/v3/index.json" />
+  </packageSources>
+</configuration>

